lib/gringotts/gateways/trexle.ex

```diff
@@ -1,56 +1,206 @@
+
+  Your Application config must look something like this:
+
+      config :gringotts, Gringotts.Gateways.Trexle,
+          adapter: Gringotts.Gateways.Trexle,
+          api_key: "Secret API key",
+          default_currency: "USD"
   """
+
   @base_url "https://core.trexle.com/api/v1/"
+  @currency "USD"
+  @email "john@trexle.com"
+  @ip_address "66.249.79.118"
+  @description "Store Purchase 1437598192"
 
   use Gringotts.Gateways.Base
   use Gringotts.Adapter, required_config: [:api_key, :default_currency]
   import Poison, only: [decode: 1]
   alias Gringotts.{Response}
 
+  @doc """
+  Authorizes your card with the given amount and returns a charge token and captured status as false in response.
+
+  ### Example
+  ```
+  iex> amount = 100
+
+  iex> card = %{
+    name: "John Doe",
+    number: "5200828282828210",
+    expiry_month: 1,
+    expiry_year: 2018,
+    cvc:  "123",
+    address_line1: "456 My Street",
+    address_city: "Ottawa",
+    address_postcode: "K1C2N6",
+    address_state: "ON",
+    address_country: "CA"
+  }
+
+  iex> options = [email: "john@trexle.com", ip_address: "66.249.79.118" , description: "Store Purchase 1437598192"]
+
+  iex> Gringotts.authorize(:payment_worker, Gringotts.Gateways.Trexle, amount, card, options)
+  ```
+  """
+
   @spec authorize(float, map, list) :: map
   def authorize(amount, payment, opts \\ []) do
     params = create_params_for_auth_or_purchase(amount, payment, opts, false)
     commit(:post, "charges", params, opts)
   end
 
+  @doc """
+  Performs the amount transfer from the customer to the merchant.
+
+  The actual amount deduction performed by Trexle using the customer's card info.
+
+  ## Example
+  ```
+  iex> card = %{
+    name: "John Doe",
+    number: "5200828282828210",
+    expiry_month: 1,
+    expiry_year: 2018,
+    cvc:  "123",
+    address_line1: "456 My Street",
+    address_city: "Ottawa",
+    address_postcode: "K1C2N6",
+    address_state: "ON",
+    address_country: "CA"
+  }
+
+  iex> options = [email: "john@trexle.com", ip_address: "66.249.79.118" ,description: "Store Purchase 1437598192"]
+
+  iex> amount = 50
+
+  iex> Gringotts.purchase(:payment_worker, Gringotts.Gateways.Trexle, amount, card, options)
+  ```
+  """
+
   @spec purchase(float, map, list) :: map
   def purchase(amount, payment, opts \\ []) do
     params = create_params_for_auth_or_purchase(amount, payment, opts)
     commit(:post, "charges", params, opts)
   end
 
+  @doc """
+  Captures a particular amount using the charge token of a pre authorized card.
+
+  The amount specified should be less than or equal to the amount given prior to capture while authorizing the card.
+  If the amount mentioned is less than the amount given in authorization process, the mentioned amount is debited.
+  Please note that multiple captures can't be performed for a given charge token from the authorisation process.
```

Comment:
Use authorization instead of authorisation

Typically British English speakers prefer 'authorise' and American English speakers prefer 'authorize', but in fact both spellings are correct 'British' usage. ... To authorize comes from the medieval Latin verb auctorizare, so some people might claim that authorize is the etymologically correct spelling.